File: LogisimPlus/LogisimPlus/logisim/std/wiring/ConstantConfigurator.cs
// ====================================================================================================
// Produced by the Free Edition of Java to C# Converter.
// To produce customized conversions, purchase a Premium Edition license:
// https://www.tangiblesoftwaresolutions.com/product-details/java-to-csharp-converter.html
// ====================================================================================================

/* Copyright (c) 2010, Carl Burch. License information is located in the
 * logisim.Main source code and at www.cburch.com/logisim/. */

namespace logisim.std.wiring
{
	using AttributeSet = logisim.data.AttributeSet;
	using BitWidth = logisim.data.BitWidth;
	using StdAttr = logisim.instance.StdAttr;
	using IntegerConfigurator = logisim.tools.key.IntegerConfigurator;

	internal class ConstantConfigurator : IntegerConfigurator
	{
		public ConstantConfigurator() : base(Constant.ATTR_VALUE, 0, 0, 0, 16)
		{
		}

		public override int getMaximumValue(AttributeSet attrs)
		{
			BitWidth width = attrs.getValue(StdAttr.WIDTH);
			int ret = width.Mask;
			if (ret >= 0)
			{
				return ret;
			}
			else
			{
				return int.MaxValue;
			}
		}

		public override int getMinimumValue(AttributeSet attrs)
		{
			BitWidth width = attrs.getValue(StdAttr.WIDTH);
			if (width.Width < 32)
			{
				return 0;
			}
			else
			{
				return int.MinValue;
			}
		}
	}

}

File: logisim/app/src/main/java/logisim/std/wiring/ConstantConfigurator.java
/* Copyright (c) 2010, Carl Burch. License information is located in the
 * logisim.Main source code and at www.cburch.com/logisim/. */

package logisim.std.wiring;

import logisim.data.AttributeSet;
import logisim.data.BitWidth;
import logisim.instance.StdAttr;
import logisim.tools.key.IntegerConfigurator;

class ConstantConfigurator extends IntegerConfigurator {
	public ConstantConfigurator() {
		super(Constant.ATTR_VALUE, 0, 0, 0, 16);
	}

	@Override
	public int getMaximumValue(AttributeSet attrs) {
		BitWidth width = attrs.getValue(StdAttr.WIDTH);
		int ret = width.getMask();
		if (ret >= 0) return ret;
		else return Integer.MAX_VALUE;
	}

	@Override
	public int getMinimumValue(AttributeSet attrs) {
		BitWidth width = attrs.getValue(StdAttr.WIDTH);
		if (width.getWidth() < 32) return 0;
		else return Integer.MIN_VALUE;
	}
}
